### PR TITLE
feat: diagnostics tab

### PR DESCRIPTION
**Pull request by eabruzzese**
This PR is based on @joeclayallday's work in #1001, although a simpler first pass. It adds a "Diagnostics" page, where a user (just Aptible organization members, for now) can generate a troubleshooting dashboard via Aptible AI.

Note that this is just a first pass to get something into the hands of CREs. Ultimately we want to render the dashboards in app-ui instead of the janky button-redirect we have here, and that's what we're doing next.

app-ui-integration

Diagnostics are also available as a tab on the app detail page:

Screenshot 2024-12-20 at 11 42 32 PM

### CODE REVIEW

File: src/ui/pages/ftux-redirects.test.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe("CreateProjectFromAppSetupPage", () => {
   });
 
   describe("when there is just an app configure operation", () => {
-    it("should redirect to status page", async () => {
+    it.skip("should redirect to status page", async () => {
```
Comment (eabruzzese) [inline src/ui/pages/ftux-redirects.test.tsx:53]: Disabled based on [this](https://aptible.slack.com/archives/C06F3FY1F0R/p1734979459496859) Slack conversation.

File: src/ui/pages/diagnostics.tsx
Comment (eabruzzese) [inline src/ui/pages/diagnostics.tsx:1]: This list view is just a mock, and isn't used yet, but will be updated in the next PR.

**Review by almathew — APPROVED** [review]

**Comment by almathew** [comment]
I'm kind of curious to check this in staging/sbx-main, but overall lgtm

File: src/ui/pages/app-detail-diagnostics.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { useParams } from "react-router";
+import { Banner, Box, Group, PermissionGate, tokens } from "../shared";
+import { DiagnosticsCreateForm } from "./diagnostics-create";
+
+export function AppDetailDiagnosticsPage() {
+  const { id = "" } = useParams();
+  const app = useSelector((s) => selectAppById(s, { id }));
+
+  return (
+    <PermissionGate scope="read" envId={app.environmentId}>
```
[the remark this answers is not in the document]
Comment (eabruzzese) [reply src/ui/pages/app-detail-diagnostics.tsx:12]: Ah, I was just following the pattern in [`app-detail-deps.tsx`](https://github.com/aptible/app-ui/blob/main/src/ui/pages/app-detail-deps.tsx#L28), which I figured was pretty similar to the permissions required to see everything in a dashboard.

[the remark this answers is not in the document]
Comment (almathew) [reply src/ui/pages/app-detail-diagnostics.tsx:12]: We should discuss access - this is something I was struggling with - but I think in the future we need to limit to anyone who can normally read metrics and logs, which is admin + deployment + ops permissions. It's not a problem right now, though.

**Pull request merged**